android/java/org/chromium/chrome/browser/app/domain/KeyringModel.java

```diff
@@ -182,15 +197,15 @@ public LiveData<AccountInfo> getSelectedAccountOrAccountPerOrigin() {
                 return mSelectedAccount;
             }
             _mSelectedAccount.setValue(null);
-            mKeyringService.getSelectedAccount(mSharedData.getCoinType(), accountAddress -> {
-                if (accountAddress == null) {
-                    mKeyringService.getKeyringInfo(
-                            getSelectedCoinKeyringId(mSharedData.getCoinType()),
-                            keyringInfo -> { updateSelectedAccountPerOriginOrFirst(keyringInfo); });
-                } else {
-                    update();
-                }
-            });
+            @CoinType.EnumType
+            int coinType = mSharedData.getCoinType();
+            if (coinType == CoinType.FIL) {
+                mKeyringService.getFilecoinSelectedAccount(BraveWalletConstants.FILECOIN_MAINNET,
```

Comment:
Avoid hard-coded `BraveWalletConstants.FILECOIN_MAINNET` value here. Need to use the selected network as it could be FILECOIN_TESTNET